### DISCUSSION TITLE
Way to get absolute path of JDL ?

### DISCUSSION THREAD

**Discussion by panta-123** [DIRACGrid/DIRAC#6699] [Q&A] [answered]
I want to get some info from JDL  while exceuting the scipt https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/WorkloadManagementSystem/Client/DownloadInputData.py. 

I am using `../jobID.jdl`  to get the jdl file location. Is there a way to get the jdl file location path without using relative path ?

**Comment by chaen** [comment]
I am not sure I really understand... This file is not a script, but it is a module used by the `JobWrapper`. And as far as I can tell, that script does not use the `JDL` directly. Could you explain what you are actually trying to do ? 

Now in general, if you want to convert a relative path to an absolute path, use `os.path.realpath` (https://docs.python.org/3.8/library/os.path.html#os.path.realpath)

**Reply by panta-123** [reply]
I want to have know the location of .jdl in wokernode, to get so information.
I am doing `../jobID.jdl` for the path in the mentioned module and then using ClassAd to extract some info.
Question is that is the another way to locate .jdl file in worker node?

**Comment by fstagni** [comment, marked as answer]
I am also not sure I fully understand, but IIUC there's the need for few notions. The DIRAC WMS isn't the easiest of the systems to understand and I think some explanations are needed.

The [JDL](https://indico.cern.ch/event/422979/contributions/1036845/attachments/892335/1256365/JDL.pdf) is an old-ish format for representing a job "in the grid" (whatever that means) but in DIRAC is not very important :)
In DIRAC there's a more powerful way: https://dirac.readthedocs.io/en/latest/AdministratorGuide/Systems/Workflow/index.html
In practice, the "workflow" above has an xml representation and a "jobDescription.xml" file is added in InputSandbox. There are methods to create the JDL starting from the XML. While you can find the XML file in the same directory of the Worker Node of where the job is allowed to run, you won't find the JDL file.
